app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -59,60 +69,18 @@ def capabilities
       [:image]
     end
 
-    def self.regions
-      [
-          ['West Europe', 'westeurope'],
-          ['Central US', 'centralus'],
-          ['South Central US', 'southcentralus'],
-          ['North Central US', 'northcentralus'],
-          ['West Central US', 'westcentralus'],
-          ['East US', 'eastus'],
-          ['East US 2', 'eastus2'],
-          ['West US', 'westus'],
-          ['West US 2', 'westus2'],
-          ['Canada Central', 'canadacentral'],
-          ['Canada East', 'canadaeast'],
-          ['Brazil South', 'brazilsouth'],
-          ['North Europe', 'northeurope'],
-          ['France Central', 'francecentral'],
-          ['France South', 'francesouth'],
-          ['UK South', 'uksouth'],
-          ['UK West', 'ukwest'],
-          ['Germany Central', 'germanycentral'],
-          ['Germany Northeast', 'germanynortheast'],
-          ['Germany West Central', 'germanywestcentral'],
-          ['Germany North', 'germanynorth'],
-          ['Switzerland North', 'switzerlandnorth'],
-          ['Switzerland West', 'switzerlandwest'],
-          ['Norway West', 'norwaywest'],
-          ['Norway East', 'norwayeast'],
-          ['East Asia', 'eastasia'],
-          ['Southeast Asia', 'southeastasia'],
-          ['Australia Central', 'australiacentral'],
-          ['Australia Central 2', 'australiacentral2'],
-          ['Australia East', 'australiaeast'],
-          ['Australia Southeast', 'australiasoutheast'],
-          ['China East', 'chinaeast'],
-          ['China North', 'chinanorth'],
-          ['China East 2', 'chinaeast2'],
-          ['China North 2', 'chinanorth2'],
-          ['Central India', 'centralindia'],
-          ['South India', 'southindia'],
-          ['West India', 'westindia'],
-          ['Japan East', 'japaneast'],
-          ['Japan West', 'japanwest'],
-          ['Korea Central', 'koreacentral'],
-          ['Korea South', 'koreasouth'],
-          ['South Africa North', 'southafricanorth'],
-          ['South Africa West', 'southafricawest'],
-          ['UAE Central', 'uaecentral'],
-          ['UAE North', 'uaenorth']
-      ]
+    def regions
+      return unless sub_id.present?
+      available_azure_locations = []
+      sdk.list_locations(sub_id).value.each do |loc|
+        location_names = []
+        location_names << loc.display_name
+        location_names << loc.name
+        available_azure_locations << location_names
+      end
```

Comment:
```suggestion
      sdk.list_locations(sub_id).value.map { |loc| [loc.display_name, loc.name] }
```
And then you can return it directly without `available_azure_locations`